packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -112,6 +112,10 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
                 }));
             }
         }, options);
+
+        this.registerRestartCommand();
```

Comment:
I think it should be available only if the server actually running, i.e. like in typescript extension right now.

We also need to remove an existing command from the typescript extension to avoid duplicate.

Reply:
> I think it should be available only if the server actually running, i.e. like in typescript extension right now.

This is what this code does, doesn't it?  `activate` is only called when starting the language client/server.

> We also need to remove an existing command from the typescript extension to avoid duplicate.

Ok.

Reply:
If the language server stops, we should remove the command, but I don't know how to make a language server stop for good.